### DISCUSSION TITLE
[Roadmap1]遇到的问题

### DISCUSSION THREAD

**Discussion by uvd** [hoh-zone/SuiStartrek#26] [General]
遇到的问题讨论

**Comment by jeffierw** [comment]
不知道怎么在discussions发图片 所以把遇到的问题和解决办法记录在notion了 有遇到这些问题的可以看看 [notion](https://jeffierw.notion.site/jeffierw/SuiStartrek-Problems-b0a25bbaefe14b679f71d0bf17e27e5a)

**Reply by upupnoah** [reply]
可以通过图床发图片hh, https://imgur.com/

**Comment by ilses1** [comment]
win11下开了全局梯子安装sui遇到的问题，执行以下代码
` cargo install --locked --git https://github.com/MystenLabs/sui.git --branch devnet sui`
安装进度到一定阶段，出现以下错误:
![image](https://github.com/movefuns/SuiStartrek/assets/61105765/ba8abf85-5a5e-4e14-ba7d-3256bb969275)
![image](https://github.com/movefuns/SuiStartrek/assets/61105765/476a11b8-b0ea-4b8c-8afa-9a0c5544b079)
![image](https://github.com/movefuns/SuiStartrek/assets/61105765/9541b2a9-227f-4e67-aea8-4fc6dca7f3af)

**Reply by dethan3** [reply]
是不是把 CPU 和 内存撑爆了？

**Comment by dethan3** [comment]
## Counter App 无法支付 Gas 
前面的步骤都做了，从前端链接到浏览器中的钱包地址也成功。
点击 `Create Counter` 无法完成交易，请问我大概是哪里出错了？
![image](https://github.com/movefuns/SuiStartrek/assets/33822639/e7fbd637-471c-4177-87e2-3e1e3d9d7250)

**Reply by dethan3** [reply]
已解决，因为 sui wallet 没有自动切换到 devnet，手动切换后就可以在前端完成交互了！Hello Sui World

**Comment by qiaopengjun5162** [comment]
sui/dapp/hello-world via ⬢ v21.2.0 via 🅒 base 
➜ sui client new-env --alias devnet --rpc https://fullnode.devnet.sui.io:443

Config file ["/Users/qiaopengjun/.sui/sui_config/client.yaml"] doesn't exist, do you want to connect to a Sui Full node server [y/N]?y
Sui Full node server URL (Defaults to Sui Devnet if not specified) : 

这里怎么设置？

**Reply by qiaopengjun5162** [reply]
sui/dapp/hello-world via ⬢ v21.2.0 via 🅒 base 
➜ sui client new-env --alias devnet --rpc https://fullnode.devnet.sui.io:443

Config file ["/Users/qiaopengjun/.sui/sui_config/client.yaml"] doesn't exist, do you want to connect to a Sui Full node server [y/N]?y
Sui Full node server URL (Defaults to Sui Devnet if not specified) : 
Select key scheme to generate keypair (0 for ed25519, 1 for secp256k1, 2: for secp256r1):
0
Generated new keypair for address with scheme "ed25519" [0x35370841d2e69b495b1e2f944a3087e4242f314e503691a00b054e0ee2a45a73]
Secret Recovery Phrase : [lend chicken payment powder glove core labor crop kidney wet brisk museum]
Environment config with name [devnet] already exists.

sui/dapp/hello-world via ⬢ v21.2.0 via 🅒 base took 6m 27.4s 
➜

**Reply by dethan3** [reply]
说明你已经有 devnet 了，用 `sui client switch --env devnet` 切换过去就能用了

**Comment by 2niuhe** [comment]
当前发布合约只能用devnet吧，testnet无法获取到gas

**Comment by 2niuhe** [comment]
目前macos上源码编译安装最新的testnet分支的sui有问题，可以安装devnet分支的作为替代

**Comment by ilses1** [comment]
https://docs.sui.io/guides/developer/first-app/build-test  官网案例打包出错
![image](https://github.com/movefuns/SuiStartrek/assets/61105765/23a84e3e-b38f-48be-b7f6-e480e6eba0a8)

**Reply by ilses1** [reply]
警告字段没有被使用，在对应的struct定义前，加上`#[allow(unused_field)]`来忽略未使用字段
![image](https://github.com/movefuns/SuiStartrek/assets/61105765/a558a4f5-baa2-4d2b-8a95-4805d20e2eaa)
即可打包成功
![image](https://github.com/movefuns/SuiStartrek/assets/61105765/847fff6a-66ad-4c27-a2df-1cc2467fb10c)
